tutorials/models_mapping/plot_1_tensor_models.py

```diff
@@ -0,0 +1,337 @@
+
+# Define the model
+mod = halfspace_val*np.ones(ind_active.sum())
+
+# We can plot a slice of the model at Y=-2.5
+fig = plt.figure(figsize=(5, 5))
+ax = fig.add_subplot(111)
+mesh.plotSlice(mod_map*mod, normal='Y', ax=ax, ind=int(mesh.nCy/2), grid=True)
+ax.set_title('Model slice at y = -2.5 m')
+plt.show()
+
+#############################################
+# Topography, a block and a vertical dyke
+# ---------------------------------------
+#
+# In this example we create a model containing a block and a vertical dyke
+# that strikes along the y direction. The utility *surface2ind_topo* is used
+# to find the cells which lie below a set of xyz points defining a surface.
+#
+
+mesh = make_example_mesh()
+
+background_val = 100.
+dyke_val = 40.
+block_val = 70.
+
+# Define surface topography as an (N, 3) np.array. You could also load a file
+# containing the xyz points
+[xx, yy] = np.meshgrid(mesh.vectorNx, mesh.vectorNy)
+zz = -3*np.exp((xx**2 + yy**2) / 75**2) + 40.
+topo = np.c_[mkvc(xx), mkvc(yy), mkvc(zz)]
+
+# Find cells below topography and define mapping
+air_val = 0.
+ind_active = surface2ind_topo(mesh, topo, 'N')
+mod_map = maps.InjectActiveCells(mesh, ind_active, air_val)
+
+# Define the model on subsurface cells
+mod = background_val*np.ones(ind_active.sum())
+ind_dyke = (mesh.gridCC[ind_active, 0] > 20.) & (mesh.gridCC[ind_active, 0] < 40.)
+mod[ind_dyke] = dyke_val
+ind_block = (
+    (mesh.gridCC[ind_active, 0] > -40.) & (mesh.gridCC[ind_active, 0] < -10.) &
+    (mesh.gridCC[ind_active, 1] > -30.) & (mesh.gridCC[ind_active, 1] < 30.) &
+    (mesh.gridCC[ind_active, 2] > -40.) & (mesh.gridCC[ind_active, 2] < 0.)
+)
+mod[ind_block] = block_val
+
+# Plot
+fig = plt.figure(figsize=(5, 5))
+ax = fig.add_subplot(111)
+mesh.plotSlice(mod_map*mod, normal='Y', ax=ax, ind=int(mesh.nCy/2), grid=True)
+ax.set_title('Model slice at y = -2.5 m')
+plt.show()
+
+
+#############################################
+# Combo Maps
+# ----------
+#
+# Here we demonstrate how combo maps can be used to create a single mapping
+# from the model to the mesh. In this case, our model consists of
+# log-conductivity values but we want to plot the resistivity. To accomplish
+# this we must take the exponent of our model values, then take the reciprocal,
+# then map from below surface cell to the mesh.
+#
+
+mesh = make_example_mesh()
+
+background_val = np.log(1./100.)
+dyke_val = np.log(1./40.)
+block_val = np.log(1./70.)
+
+# Define surface topography
+[xx, yy] = np.meshgrid(mesh.vectorNx, mesh.vectorNy)
+zz = -3*np.exp((xx**2 + yy**2) / 75**2) + 40.
+topo = np.c_[mkvc(xx), mkvc(yy), mkvc(zz)]
+
+# Find cells below topography
+air_val = 0.
+ind_active = surface2ind_topo(mesh, topo, 'N')
+active_map = maps.InjectActiveCells(mesh, ind_active, air_val)
+
+# Define the model on subsurface cells
+mod = background_val*np.ones(ind_active.sum())
+ind_dyke = (mesh.gridCC[ind_active, 0] > 20.) & (mesh.gridCC[ind_active, 0] < 40.)
+mod[ind_dyke] = dyke_val
+ind_block = (
```

Comment:
Can we use `ModelBuilder.PolygonInd` here too?

Reply:
I actually did some of this type of stuff deliberately. I know we are spending time developing great utilities. But I wanted to demonstrate that you could find the indicies in this way and change model parameters in this fashion.